core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -290,12 +299,121 @@ public ClientFactoryBuilder tlsNoVerifyHosts(String... insecureHosts) {
         return this;
     }
 
+    /**
+     * Configures SSL or TLS for client certificate authentication with the specified {@code keyCertChainFile}
+     * and cleartext {@code keyFile}.
+     */
+    @Override
+    public ClientFactoryBuilder tls(File keyCertChainFile, File keyFile) {
```

Comment:
optional nit; What do you think of adding a variant specially for clients since a private key isn't necessary if mtls isn't used?

```suggestion
    public ClientFactoryBuilder tls(File keyCertChainFile) {
```

Reply:
> Did you mean trusted certificates that verify the remote endpoint's certificate?

Yes. From just looking at the API set, it seems to imply that `keyFile` is always necessary for client-side (which it isn't).
Having said this, I guess it's fine to omit and add later since I don't think there will be many use-cases for it

Reply:
Offline discussion) Decided that it might be better to add an API for this if there is any need for it in the future.